### PR TITLE
Generalize for renderTextarea

### PR DESCRIPTION
**Pull request by stevemk14ebr**
Few changes, closes https://github.com/uiwjs/react-md-editor/issues/193:

* renderTextarea's onChange event has been simplified and generalized to accept just the changes text string. It will handle its own dispatch and state management
* the markdown pre element and transparent textarea text have been removed. There were slight misalignments between the overlay and the textarea depending on the child content. I think this is actually the cause of https://github.com/uiwjs/react-md-editor/issues/147. The user can use rehype syntax plugins on the markdown preview instead of the buggy transparent overlay technique

Example of new renderTextarea:
```
renderTextarea={(props, { handleChange }) => {
                    let { value, ...otherProps } = props;
                   
                    // html strings must be used, react JSX would be invalid since React doesn't manage these nodes
                    let texts = value.split("\n").map((item) => {
                        // each line is enclosed with a span already, each space subdivides from the middle
                        let split = item.replaceAll(" ", "</span><span> ");
                        return "<span>" + split + "\n</span>";
                    }).join("");
                    return (
                        <div {...otherProps} style={{ whiteSpace: 'pre' }} spellCheck={true} contentEditable={true} suppressContentEditableWarning={true} onInput={(e) => {
                            setCaretPos(Cursor.getCurrentCursorPosition(e.target));
                            handleChange(e.target.innerText);
                        }}
                            dangerouslySetInnerHTML={{ __html: texts }}
                        />
                    )
                }}
```

dangerouslySetInnerHTML is used because React is not allowed to manage the children since user edits could delete them at any time. I put spans between each word for drag and drop logic to be able to easily find word boundaries.

### CODE REVIEW

**Comment by jaywcjlove** [comment]
@stevemk14ebr You need to resolve some errors. https://vercel.com/wcj/react-md-editor/Ck2bwAcNgjAdmMSBATkAP9VSpc2J

```
components/TextArea/index.tsx(6,22): error TS2307: Cannot find module './Markdown' or its corresponding type declarations.
```

File: src/components/TextArea/index.less [hunk 1]
```diff
@@ -57,12 +57,10 @@
       height: 100%;
       width: 100%;
       resize: none;
-      color: inherit;
       overflow: hidden;
       outline: 0;
       padding: 10px;
       -webkit-font-smoothing: antialiased;
-      -webkit-text-fill-color: transparent;
```
Comment (jaywcjlove) [inline src/components/TextArea/index.less:65]: This cannot be deleted, and the highlighting of the editing area is invalid.

Comment (stevemk14ebr) [reply src/components/TextArea/index.less:65]: The overlay as it current is implemented does not work for all content. Try my example with spans in the content using an editable div and you'll see that the transparent text of the div does not align properly with the visible text of the pre. I have therefore removed the entire overlay, and make the div/textarea text non-transparent.

The user can instead do syntax highlighting in the markdown preview pane by adding their own highlighting pass to the plugins.

Comment (jaywcjlove) [reply src/components/TextArea/index.less:65]: Your idea will invalidate the editor code highlighting.

Comment (stevemk14ebr) [reply src/components/TextArea/index.less:65]: correct highlighting in the textarea/div side is removed. Is this not acceptable?

I was unable to fix the overlay alignment with css, perhaps you could try? If it is fixable it won't need to be removed. Just use a contenteditable div and then out spans inside it

Comment (jaywcjlove) [reply src/components/TextArea/index.less:65]: Add a new API, to be forward compatible.

**Comment by stevemk14ebr** [comment]
i'll re-open this later

**Pull request closed without merging**